### ISSUE TITLE
ENH: confusing name for 'copy' argument in to_numpy() method

### ISSUE THREAD

**Issue by Abdelgha-4** [pandas-dev/pandas#43165]
I found the following behavior a bit unexpected from the name of the argument.

> The default (`copy=False`) may return a copy instead of a view. `copy=True` always returns a copy, but `copy=False` does not always return a view.

I think changing the argument name to something like `always_copy` would be more clear about what it is doing.

**Comment by attack68** [comment]
-1, whilst what you are proposing might be clearer, I consider the existing docs explicitly clear enough. Perhaps expanding:

> `copy=False` does not always return a view and still may return a copy depending upon object type.

**Comment by mzeitlin11** [comment]
Agreed with @attack68, this conditional copying mirrors `numpy` behavior as well, for example in https://numpy.org/doc/stable/reference/generated/numpy.array.html

> If true (default), then the object is copied. Otherwise, a copy will only be made if __array__ returns a copy, if obj is a nested sequence, or if a copy is needed to satisfy any of the other requirements (dtype, order, etc.).

**Comment by jorisvandenbossche** [comment]
Small note that in numpy they have been discussing changing the arguments for this keyword, cfr https://github.com/numpy/numpy/pull/19173 (and mailing list discussions), to be able to distinguish those cases of "never coying" vs "only copy when needed"

**Comment by Abdelgha-4** [comment]
NumPy discussion about this has been ended in [numpy/numpy#19173](https://github.com/numpy/numpy/pull/19173), so I guess we could make a decision about this issue now ?
